### PR TITLE
fix: removed condition that prevented blocking api

### LINKED ISSUE #3: Rate limiter not blocking requests when using throttle middleware
### Package version

1.0.1

### Describe the bug

I tried to implement rate limiting for an API which will be allowed to call only 1 time every 20 seconds but if the limit exceeds, it should be blocked for 20 minutes. 

I have added two API endpoints in the sample repo:
1. GET /test/using-middleware
2. POST /test/using-limiter-manager

In the `GET /test/using-middleware` route, The API endpoint throws an error if it is called more than 1 time, however, after 20 seconds, I am able to call the API again, instead of the API endpoint getting blocked for 20 minutes.

In the `start/limiter.ts file`, I have defined the rule as follows:
```ts
import { Limiter } from '@adonisjs/limiter/build/services'

export const { httpLimiters } = Limiter.define('global', () => {
  return Limiter.allowRequests(1).every('20 second').blockFor('2 min')
})
```

In the `start/routes.ts` file, I have added the global middleware as follows:
```ts
Route.get('/test/using-middleware', async () => {
  return { hello: 'world' }
}).middleware('throttle:global')
```

In the `POST /test/using-limiter-manager` route, The API endpoint is getting blocked properly. I have implemented blocking for this route as specified in https://docs.adonisjs.com/guides/rate-limiting#login-endpoint-brute-force-protection.

I am using `redis` as a store for rate-limiting data. 

For `GET /test/using-middleware` route, I am getting the below response in Redis CLI:
```
127.0.0.1:6379> TTL rlflx:global_127.0.0.1
(integer) 18
```

For `POST /test/using-limiter-manager` route, I am getting the below response in Redis CLI:
```
127.0.0.1:6379> TTL rlflx:127.0.0.1
(integer) 118
```

### Reproduction repo

https://github.com/shahidshahcredilio/rate-limiter-test

### PR DESCRIPTION
**Pull request by ghost**
### 🔗 Linked issue

#3 

### ❓ Type of change

- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The rate limiter not blocking requests when using throttle middleware but works when we use the limit manager API. The change removes an `if` condition where we are returning from the method instead of incrementing the counter by calling the consume method. Due to this, the catch block in the consume method is not executed and the `retryAfter` key is not updated with the `blockDuration` time and the APIs don't get blocked.

Resolves #3 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by thetutlage** [comment]
Have you made sure that removing the existing code does not break anything else? I believe the conditional was in place for a reason and also the `abort` method calls the `limitedExceededCallback`, which will not be called anymore after your change

**Comment by thetutlage** [comment]
I think the better fix will be to tweak the condition from `<= 0` to `< 0`. In this way, if the remaining points are `0`, then the `consume` method will be called and hence the block logic will trigger.

Also, it will be nice to have a test for the same.

**Comment by ghost** [comment]
> Have you made sure that removing the existing code does not break anything else? I believe the conditional was in place for a reason and also the `abort` method calls the `limitedExceededCallback`, which will not be called anymore after your change

Yes, I have tested the code and it is working as expected.

After removing the condition, the `limitedExceededCallback` is called in the `catch` block below (line 51)

**Comment by ghost** [comment]
> I think the better fix will be to tweak the condition from `<= 0` to `< 0`. In this way, if the remaining points are `0`, then the `consume` method will be called and hence the block logic will trigger.
> 
> Also, it will be nice to have a test for the same.

I will check this change, also, I will try to add tests as well, I had a problem configuring Redis, Postgres and MySQL using docker for testing.

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,13 @@
+REDIS_HOST=
```
Comment (shahidshahcredilio) [inline .env.example:1]: I have added example file for ease of setup

**Comment by ghost** [comment]
Can anyone please review the PR?

**Comment by thetutlage** [comment]
I see you have changed too many code paths in the PR. You have abstracted some types that are used during tests to their own file and also changed the markdown files.

Now, these changes might be relevant, but they are not relevant for the actual fix and makes it harder for me to focus on the actual change.

Please, keep the PR as simple as possible. You can create another PR for unrelated changes

**Comment by ghost** [comment]
I have tried to simplify the PR, please let me know if anything is needed from my end.

Thanks.

**Comment by thetutlage** [comment]
Looks great now! Thanks for keeping the patience and making necessary changes.

**Comment by mastermunj** [comment]
Hi @thetutlage, please help release these changes.

**Pull request merged**